src/network/local_socket_reply.cpp

```diff
@@ -80,6 +80,23 @@ QNetworkReply::NetworkError statusCodeFromHttp(int httpStatusCode)
 
     return code;
 }
+
+QList<QByteArray> split_http_lines(const QByteArray& reply_data)
+{
+    QList<QByteArray> list;
+    int start{0};
+    int end;
+
+    while ((end = reply_data.indexOf("\r\n", start)) != -1)
+    {
+        list.append(reply_data.mid(start, end - start));
+        start = end + 2;
+    }
+
+    list.append(reply_data.mid(start).split('\n').front());
+
+    return list;
+}
```

Comment:
Why not [`QByteArray::split()`](https://doc.qt.io/qt-5/qbytearray.html#split)?

Reply:
Because we need to split on `\r\n` first and `QByteArray::split()` can only split on a single character.

Reply:
Oh indeed. I looked at `QString::split()` first and it had `QString sep`, and no mention that it's meant to be a single character.

But yeah, `QByteArray::split()` has `char sep`…